### PR TITLE
feat: enhance user experience

### PR DESCRIPTION
**Pull request by benoitf**
- open terminal by default (if no terminal is there)
- add the remote customization: text, icon, commands available when clicking on the bottom left remote

https://user-images.githubusercontent.com/436777/151932251-d30da57a-27f2-4198-b697-92e109bf5d75.mp4

### CODE REVIEW

File: code/extensions/che-on-start/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Eclipse Che Port for Visual Studio Code
```
Comment (azatsarynnyy) [inline code/extensions/che-on-start/README.md]: Please, update the readme

File: code/extensions/che-on-start/package.nls.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"displayName": "Eclipse Che OnStart",
+	"description": "OnStart actions like project automatically at startup if no project is provided"
```
Comment (azatsarynnyy) [inline code/extensions/che-on-start/package.nls.json]: Could you please rephrase it a bit, as it's hard to understand the extension's description

File: code/extensions/che-remote/__mocks__/vscode.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/* eslint-disable header/header */
```
Comment (azatsarynnyy) [inline code/extensions/che-remote/__mocks__/vscode.ts]: Why do we need that rule here?

File: code/extensions/che-on-start/src/extension.ts [hunk 4]
```diff
@@ -13,7 +13,24 @@
...
+  const output = vscode.window.createOutputChannel('Che OnStart');
+  openTerminalIfNone(output);
+  openProjectIfNone(output);
+
+}
+
+async function openTerminalIfNone(output: vscode.OutputChannel): Promise<void> {
+  // open a new terminal if there is none to always see a terminal
+  if (vscode.window.terminals.length === 0) {
+    output.appendLine('Opening a new terminal....');
```
Comment (azatsarynnyy) [inline code/extensions/che-on-start/src/extension.ts:32]: `output` is declared in another function - `activate`.
What `output` does it use here?

Comment (benoitf) [reply code/extensions/che-on-start/src/extension.ts:32]: it's using the output given in parameter initialized in activate method ?

File: code/extensions/che-remote/__mocks__/vscode.ts (hunk 3 above)
Comment (benoitf) [reply code/extensions/che-remote/__mocks__/vscode.ts]: we have the rule in every che files else it tries to validate with Microsoft license

Comment (azatsarynnyy) [reply code/extensions/che-remote/__mocks__/vscode.ts]: I see. Thanks for the explanation!

Comment (benoitf) [reply code/extensions/che-remote/__mocks__/vscode.ts]: but I'll move this line after the header
At first I thought it was only working at the beginning of the file

And I think that probably we could tweak linter rule to ignore the che-* extensions

File: code/extensions/che-on-start/src/extension.ts (hunk 4 above)
Comment (azatsarynnyy) [reply code/extensions/che-on-start/src/extension.ts:32]: ahh... I overlooked it's passed through the parameter. Sorry for the noise )

File: code/extensions/che-on-start/README.md (hunk 1 above)
Comment (benoitf) [reply code/extensions/che-on-start/README.md]: updated

File: code/extensions/che-on-start/package.nls.json (hunk 2 above)
Comment (benoitf) [reply code/extensions/che-on-start/package.nls.json]: updated

File: code/extensions/che-remote/__mocks__/vscode.ts (hunk 3 above)
Comment (benoitf) [reply code/extensions/che-remote/__mocks__/vscode.ts]: moved the line to the bottom

**Comment by benoitf** [comment]
ok, so opening a default terminal by default is great but needs to be disabled in unit test/smoke test mode else we had random failures ! :-)

**Pull request merged**